physical/raft/raft.go

```diff
@@ -526,6 +578,17 @@ func (b *RaftBackend) Initialized() bool {
 	return init
 }
 
+func (b *RaftBackend) SetMetricsSink(sink *metricsutil.ClusterMetricSink) {
```

Comment:
Is there a reason we need a setter, or could this be an argument to NewRaftBackend?

Reply:
This is likely just an artifact of my ignorance. Given the way the [physical backends](https://github.com/hashicorp/vault/blob/master/command/commands.go#L135-L164) are setup, and the way those [factories are retrieved](https://github.com/hashicorp/vault/blob/master/command/server.go#L841) and [used](https://github.com/hashicorp/vault/blob/master/command/server.go#L869), I wasn't sure how to modify the arguments being passed into `NewRaftBackend` without messing all of that up. The config that's passed in there comes from the server's storage config and I didn't see how to plumb the metrics sink into that config, since I thought that came from the `storage` stanza in the vault config file.

I'm definitely open to suggestions though, if this isn't the right way to do this.

Reply:
No, that's a good reason I forgot about.